Assets/Scripts/DDAA/ReloadDDAA.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public sealed class ReloadDDAA
+{
+    // --------------------------------- //
+    // Singleton related implementation //
+    //----------------------------------//
+
+    private static ReloadDDAA instance = null;
+    private static readonly object padlock = new object();
+
+    private ReloadDDAA() { }
+
+    public static ReloadDDAA Instance
+    {
+        get
+        {
+            lock (padlock)
+            {
+                if (instance == null)
+                {
+                    instance = new ReloadDDAA();
+                }
+                return instance;
+            }
+        }
+    }
+
+    // NOTE!!! 
+    // All of the parameters below are the ones to change, when adjusting the DDA (unless there's a bug)
+
+    //static parameters
+    private static readonly float initialReloadPoint = 1f;
+    private static readonly float minReloadTime = 2f;
+    private static readonly float dpgContribution = 0.5f;
```

Comment:
what is dpg?

Reply:
Difficulty Point Global. It's explained in the report as well and I'll explain this later